python/plugins/processing/algs/gdal/gdal2xyz.py

```diff
@@ -36,6 +37,11 @@
 class gdal2xyz(GdalAlgorithm):
     INPUT = 'INPUT'
     BAND = 'BAND'
+    if GdalUtils.version() > 3060200:
```

Comment:
The standard way to handle GDAL version dependent functionality is to ALWAYS add the parameters, but then raise a QgsProcessingException at runtime if the option is set but the GDAL version is too low. This approach means that users will get a user-friendly explanatory message as to why the parameter isn't available in their QGIS version (as opposed to just a missing option), and also ensures that models can be safely shared across different versions and that useful errors are raised if the operation of the model would vary depending on GDAL version...

Reply:
Thanks Nyall, good point, I'll address. 

I note that there's a version check in [gdalcalc.initAlgorithm`](https://github.com/qgis/QGIS/blob/master/python/plugins/processing/algs/gdal/gdalcalc.py#L148) so `self.addParameter` doesn't get called for `EXTENT` and `EXTENT_OPT` but no version check around the declaration of `EXTENT` and `EXTENT_OPT` class properties.  

As implemented this way, the options aren't shown in the Processing GUI but the parameters can still be passed in e.g. by a processing.run call resulting in the appropriate QgsProcessingException at runtime if GDAL version check fails.  

Is this the way you prefer it handled in this PR, or would you like the version check removed completely, except for the runtime check?

Reply:
@nyalldawson version checks removed except for runtime and tests.